platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,272 @@
+
+  private static final int SCHEDULER_INTERVAL = 30;
+
+  private static final int FILE_BACKED_STREAM_THRESHOLD = 1000000;
+
+  static final String CRL_PROPERTY_KEY = Merlin.OLD_PREFIX + Merlin.X509_CRL_FILE;
+
+  static final String DEM_CRL = "/localCrl.crl";
+
+  static final String PEM_CRL = "/localCrl.pem";
+
+  @VisibleForTesting
+  static String issuerEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/encryption.properties").getPath();
+
+  @VisibleForTesting
+  static String issuerSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/issuer/signature.properties").getPath();
+
+  @VisibleForTesting
+  static String serverEncryptionPropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/encryption.properties").getPath();
+
+  @VisibleForTesting
+  static String serverSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/signature.properties").getPath();
+
+  @VisibleForTesting
+  static String crlFileLocation = new AbsolutePathResolver("etc/keystores").getPath();
+
+  private final ClientFactoryFactory factory;
+
+  private final ScheduledExecutorService scheduler;
+
+  private String crlLocationUrl;
+
+  private boolean crlByUrlEnabled;
+
+  private ScheduledFuture<?> handle;
+
+  public CrlGenerator(ClientFactoryFactory factory) {
+    this.factory = factory;
+    this.scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("crlThread"));
+  }
+
+  /** Pulls down the CRL file provided via URL and writes it locally. */
+  @Override
+  public synchronized void run() {
+
+    if (!crlByUrlEnabled || crlLocationUrl == null || !crlLocationUrl.startsWith(HTTPS)) {
+      return;
+    }
+
+    FileBackedOutputStream fileBackedOutputStream = null;
+
+    try {
+      SecureCxfClientFactory cxfClientFactory =
+          factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+      WebClient client = cxfClientFactory.getWebClient();
+      Response response = client.get();
+
+      Object entity = response.getEntity();
+      if (entity instanceof InputStream) {
+
+        // Read the response content and get the byte source
+        ByteSource byteSource;
+        try (InputStream entityStream = (InputStream) response.getEntity()) {
+          fileBackedOutputStream = new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
+          IOUtils.copy(entityStream, fileBackedOutputStream);
+          byteSource = fileBackedOutputStream.asByteSource();
+          fileBackedOutputStream.close();
+        }
+
+        // Determine the file extension
+        File crlFile;
+        String localCrlPath;
+        try (InputStream inputStream = byteSource.slice(0, 300).openStream()) {
+          if (IOUtils.toString(inputStream, StandardCharsets.UTF_8).contains("-----BEGIN")) {
+            localCrlPath = crlFileLocation + PEM_CRL;
```

Comment:
✏️ `localCrlPath` is not used anywhere else, simply set the `crlFile`

Reply:
It's used in the next try to set the property files but I can just get the path from the crlFile. 👍

Reply:
I had missed that ... my bad 😉